packages/shared/src/components/profile/SquadsList.tsx

```diff
@@ -0,0 +1,204 @@
+import request from 'graphql-request';
+import { SourceMember, SourceMemberRole } from '../../graphql/sources';
+import { largeNumberFormat } from '../../lib/numberFormat';
+import { Image } from '../image/Image';
+import SquadMemberBadge from '../squads/SquadMemberBadge';
+import { CardLink } from '../cards/Card';
+import PlusIcon from '../icons/Plus';
+import { ButtonSize, ButtonVariant } from '../buttons/common';
+import { Button } from '../buttons/ButtonV2';
+import {
+  useJoinSquad,
+  useToastNotification,
+  useViewSize,
+  ViewSize,
+} from '../../hooks';
+import { labels } from '../../lib';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { graphqlUrl } from '../../lib/config';
+import AuthContext from '../../contexts/AuthContext';
+import {
+  ProfileV2,
+  PUBLIC_SOURCE_MEMBERSHIPS_QUERY,
+} from '../../graphql/users';
+import { Connection } from '../../graphql/common';
+import { AuthTriggers } from '../../lib/auth';
+import { webappUrl } from '../../lib/constants';
+
+export interface SquadsListProps {
+  memberships?: Connection<SourceMember>;
+  userId: string;
+}
+
+function SquadItem({
+  membership,
+  loading,
+}: {
+  membership: SourceMember;
+  loading?: boolean;
+}): ReactElement {
+  const { user, showLogin } = useContext(AuthContext);
+  const router = useRouter();
+  const squad = membership.source;
+  const showJoin = !squad.currentMember?.role && !loading;
+  const { displayToast } = useToastNotification();
+
+  const { mutateAsync: joinSquad, isLoading } = useMutation(
+    useJoinSquad({ squad }),
+    {
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+      onSuccess: () => router.push(squad?.permalink),
+    },
+  );
+
+  const onJoin = async () => {
+    if (!user) {
+      showLogin({
+        trigger: AuthTriggers.JoinSquad,
+        options: {
+          onLoginSuccess: joinSquad,
+          onRegistrationSuccess: joinSquad,
+        },
+      });
+
+      return;
+    }
+
+    await joinSquad();
+  };
+
+  return (
+    <div
+      className="flex relative items-center p-2 first:ml-4 tablet:first:ml-0 w-40 tablet:w-auto bg-theme-float hover:bg-theme-hover rounded-2xl"
+      data-testid={squad.id}
+    >
+      <Link href={squad.permalink} prefetch={false} passHref>
+        <CardLink />
+      </Link>
+      <div className="flex overflow-hidden flex-col flex-1">
+        <div className="flex gap-2 items-center">
+          <Image
+            src={squad.image}
+            alt={squad.name}
+            className="w-8 h-8 rounded-full"
+          />
+          <div className="flex flex-col flex-1">
+            <div className="overflow-hidden font-bold whitespace-nowrap typo-caption1 text-ellipsis">
+              {squad.name}
+            </div>
+            <div className="overflow-hidden whitespace-nowrap text-theme-label-quaternary typo-caption2 text-ellipsis">
+              @{squad.handle}
+            </div>
```

Comment:
These should be divs or can we get away with paragraphs?

Reply:
Span perhaps?